packages/gatsby/cache-dir/query-result-store.js

```diff
@@ -115,42 +115,37 @@ export class PageQueryStore extends React.Component {
   }
 }
 
-export class StaticQueryStore extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      staticQueryData: { ...getStaticQueryResults() },
-    }
-  }
+export const StaticQueryStore = ({ children }) => {
+  const [staticQueryUpdateDate, setstaticQueryUpdateDate] = React.useState(
+    Date.now()
+  )
+  const isFirstRender = React.useRef(false)
 
-  handleMittEvent = () => {
-    this.setState({
-      staticQueryData: { ...getStaticQueryResults() },
-    })
+  const handleMittEvent = () => {
+    staticQuerySingleton.set(getStaticQueryResults())
+    setstaticQueryUpdateDate(Date.now())
   }
 
-  componentDidMount() {
-    ___emitter.on(`staticQueryResult`, this.handleMittEvent)
-    ___emitter.on(`onPostLoadPageResources`, this.handleMittEvent)
-  }
+  const handleEvent = React.useCallback(handleMittEvent, [])
 
-  componentWillUnmount() {
-    ___emitter.off(`staticQueryResult`, this.handleMittEvent)
-    ___emitter.off(`onPostLoadPageResources`, this.handleMittEvent)
-  }
+  React.useEffect(() => {
+    ___emitter.on(`staticQueryResult`, handleMittEvent)
+    ___emitter.on(`onPostLoadPageResources`, handleMittEvent)
 
-  shouldComponentUpdate(nextProps, nextState) {
-    // We want to update this component when:
-    // - static query results changed
+    return () => {
+      ___emitter.off(`staticQueryResult`, handleMittEvent)
+      ___emitter.off(`onPostLoadPageResources`, handleMittEvent)
+    }
+  }, [handleEvent])
 
-    return this.state.staticQueryData !== nextState.staticQueryData
+  if (!isFirstRender.current) {
+    staticQuerySingleton.set(getStaticQueryResults())
+    isFirstRender.current = true
   }
 
-  render() {
-    return (
-      <StaticQueryContext.Provider value={this.state.staticQueryData}>
-        {this.props.children}
-      </StaticQueryContext.Provider>
-    )
-  }
+  // React.Children.toArray doesn't traverse into fragments (see https://github.com/facebook/react/issues/6889)
+  // But this is fine since we only care about the next level of nesting anyways
+  return React.Children.toArray(children).map(child =>
```

Comment:
What's wrong with this fn?
```suggestion
  return React.Children.map(children, child =>
```

Reply:
It errors then with:

```
React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: <PartytownProvider />. Did you accidentally export a JSX literal instead of a component?
```

Reply:
what happens when you log out child?

Reply:
One gets `<PartytownProvider />`